plugins/logitech-tap/fu-logitech-tap-hdmi-device.c

```diff
@@ -0,0 +1,485 @@
+/*
+ * Copyright (c) 1999-2023 Logitech, Inc.
+ * Copyright (C) 2023 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include <linux/types.h>
+#include <linux/usb/video.h>
+#include <linux/uvcvideo.h>
+#ifdef HAVE_IOCTL_H
+#include <sys/ioctl.h>
+#endif
+
+#include <string.h>
+
+#include "fu-logitech-tap-common.h"
+#include "fu-logitech-tap-hdmi-device.h"
+
+#define FU_LOGITECH_TAP_HDMI_DEVICE_IOCTL_TIMEOUT 5000 /* ms */
+#define XU_INPUT_DATA_LEN			  8
+
+/* 2 byte for get len query */
+#define kDefaultUvcGetLenQueryControlSize 2
+
+const guchar kLogiTapCameraVersionSelector = 1;
+const guchar kLogiTapUvcXuAitCustomCsGetMmpResult = 5;
+
+const guchar kLogiTapHdmiVerSetData = 0x0B;
+
+const guchar kLogiUnitIdVidCapExtension = 0x06;
+const guchar kLogiHdmiVerGetSelector = 2;
+
+const guchar kLogiTapAitSetMmpCmdFwBurning = 0x01;
+const guchar kLogiTapVideoAitInitiateSetMMPData = 1;
+const guint kLogiDefaultImageBlockSize = 32;
+const guchar kLogiUvcXuAitCustomCsSetFwData = 0x03;
+
+const guchar kLogiTapUvcXuAitCustomCsSetMmp = 4;
+const guint kLogiDefaultAitSleepIntervalMs = 1000;
+
+/* when finalize Ait, max polling duration is 120sec */
+const guint kLogiDefaultAitFinalizeMaxPollingDurationMs = 120000;
+const guchar kLogiDefaultAitSuccessValue = 0x00;
+const guchar kLogiDefaultAitFailureValue = 0x82;
+
+struct _FuLogitechTapHdmiDevice {
+	FuLogitechTapDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuLogitechTapHdmiDevice, fu_logitech_tap_hdmi_device, FU_TYPE_LOGITECH_TAP_DEVICE)
+
+static gboolean
+fu_logitech_tap_hdmi_device_query_data_size(FuLogitechTapHdmiDevice *self,
+					    guchar unit_id,
+					    guchar control_selector,
+					    guint16 *data_size,
+					    GError **error)
+{
+	guint8 size_data[kDefaultUvcGetLenQueryControlSize] = {0x0};
+	struct uvc_xu_control_query size_query;
+	size_query.unit = unit_id;
+	size_query.selector = control_selector;
```

Comment:
you can set these directly, e.g. `struct uvc_xu_control_query size_query = { .unit = unit_id, ` etc

Reply:
Fixed